python/tskit/tables.py

```diff
@@ -2602,3 +2602,52 @@ def subset(self, nodes, record_provenance=True):
             self.provenances.add_row(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
+
+    def union(
+        self,
+        other,
+        node_mapping,
+        check_shared_equality=True,
+        add_populations=True,
+        record_provenance=True,
+    ):
+        """
+        Modifies the table collection in place by adding the non-shared
+        portions of ``other`` to itself. To perform the node-wise union,
+        the method relies on a ``node_mapping`` array, that maps nodes in
+        ``other`` to its equivalent node in ``self`` or ``tskit.NULL`` if
+        the node is exclusive to ``other``. See :meth:`TreeSequence.union` for a more
+        detailed description.
+
+        :param TableCollection other: Another table collection.
+        :param list node_mapping: An array of node IDs that relate nodes in
+            ``other`` to nodes in ``self``: the k-th element of ``node_mapping``
+            should be the index of the equivalent node in ``self``, or
+            ``tskit.NULL`` if the node is not present in ``self`` (in which case it
+            will be added to self). This
+            may be a numpy array (or array-like) object (dtype=np.int32).
```

Comment:
you don't need to specify the dtype here, thanks to the `util.safe_np_int_cast( )` call below.